parts/linux/cloud-init/artifacts/labels.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Update Labels for this Kubernetes node
+
+set -euo pipefail
+
+echo "updating labels for ${HOSTNAME}"
+
+for kubelet_label in $(echo $KUBELET_NODE_LABELS | sed "s/,/ /g")
```

Comment:
we don't have to break this into a loop, kubectl label supports multiple labels in one shot. this will reduce calls to apiserver and also simplify the code here.

tony@Azure:~$ kubectl label nodes aks-nodepool1-24689654-vmss000001 --overwrite x=y z=j
node/aks-nodepool1-24689654-vmss000001 labeled

Also consider printing the result of the command execution like the one in my example, that makes it easy to debug.

Reply:
I do think it is worth doing a bit optimization here so that we don't always write to api-server. We first do a kubectl get labels and get those we care, do a comparison to our goal state, and only overwrite the value in api-server if it's not in goal state. 

This will effectively converts most of the write operation to read operation. When all the nodes keep writing , I think it will adds burden to api-server and etcd and will consume more resources that is paid by us (and may even cause resource contention issues). 

I don't know how much impact will we add here by keep reading from all the nodes(I think it should be fine as I see you are reducing the frequency of running this to 5 min), but it's good to consider it(especially for large clusters). Better to give it a try on a real cluster

Reply:
"Also consider printing the result of the command execution like the one in my example, that makes it easy to debug."
Not sure exactly what you mean by this. Does the output for the commands not automatically get logged? Do you mean print the state of the labels after applying them?

Reply:
Removed the loop, and added the optimization, but it should be tested

Reply:
I added the optimization, and plan to keep it even though it was mainly needed when the service was running continuously.